### PR TITLE
Module one issues fixed

### PR DESCRIPTION
**Pull request by SwapnilQA**

### CODE REVIEW

File: .metadata/.log [hunk 1]
```diff
@@ -171,3 +171,69 @@ This warning can be switched off on the Team > Git > Confirmations and Warnings
...
+!MESSAGE Using platform encoding (Cp1252 actually) to copy filtered resources, i.e. build is platform dependent!
+
+!ENTRY org.eclipse.m2e.logback.appender 2 0 2023-06-20 13:00:52.365
+!MESSAGE Using platform encoding (Cp1252 actually) to copy filtered resources, i.e. build is platform dependent!
+
+!ENTRY org.eclipse.m2e.logback.appender 2 0 2023-06-20 16:11:40.338
+!MESSAGE Using platform encoding (Cp1252 actually) to copy filtered resources, i.e. build is platform dependent!
+
+!ENTRY org.eclipse.m2e.logback.appender 2 0 2023-06-20 16:11:40.348
+!MESSAGE Using platform encoding (Cp1252 actually) to copy filtered resources, i.e. build is platform dependent!
```
Comment (ranbharerp15) [inline .metadata/.log:239]: add these in gitignore to avoid committing such files

File: Java_Training/src/main/java/Teacher.java [hunk 2]
```diff
@@ -1,39 +1,50 @@
...
+		this.department = department;
 	}
 	
-	public String getCourse(String course) {
-		return this.course;
+	/**
+	 * This method will assign a value to specialization
+	 * @param specialization
+	 */
+	public void setSpecialization(String specialization) {
```
Comment (ranbharerp15) [inline Java_Training/src/main/java/Teacher.java:32]: java doc can be avoided on simple getter/setter methods

File: Java_Training/src/main/java/PalindromeString.java [hunk 3]
```diff
@@ -7,21 +7,32 @@ public static void main(String[] args) {
...
 	}
 
+	/**
+	 * This method will reverse the string
+	 * @param str
+	 * @return
+	 */
 	public static String stringReverse(String str) {
-		String tmp="";
+		StringBuffer tmp = new StringBuffer("");
```
Comment (ranbharerp15) [inline Java_Training/src/main/java/PalindromeString.java:20]: StringBuilder is enough here. StringBuffer has synchronized methods for thread safety. Using it here is additional overhead.

**Review by ranbharerp15 — CHANGES_REQUESTED** [review]

**Comment by SwapnilQA** [comment]
Fixing the issues

**Pull request closed without merging**